src/library/scala/io/Source.scala

```diff
@@ -167,6 +167,10 @@ object Source {
 
   def fromInputStream(is: InputStream)(implicit codec: Codec): BufferedSource =
     createBufferedSource(is, reset = () => fromInputStream(is)(codec), close = () => is.close())(codec)
+
+  def fromClassPath(resource: String, classLoader: ClassLoader = ClassLoader.getSystemClassLoader)(implicit codec: Codec): BufferedSource =
```

Comment:
agree it should be the context classloader, and I like `fromResource`. (as a bonus, you won't have to remember whether to capitalize the P in classpath; there isn't a lot of consistency out there on that point.)